go/tools/bazel/runfiles/resolver_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runfiles
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func fakeGetenv(env map[string]string) func(string) string {
+	return func(key string) string {
+		return env[key]
+	}
+}
+
+func TestManifestRunfiles(t *testing.T) {
```

Comment:
I think it would be better to test against `bazel.Runfile` directly instead of the `Resolver` interface. I'd prefer `Resolver` not be a public API, and it's better to test against public APIs whenever possible.

Reply:
Alright, I've tried to make the tests use the public API. The memoization of the initialization adds a complication but otherwise it's a straightforward change. Let me know if I should do more to make this agnostic.